capabilityreceiver/pkg/chplmapper/capabilityinfo.go

```diff
@@ -26,9 +26,50 @@ var fluffWords = []string{
 	"corporation.",
 }
 
+type details struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+type ChplEndpointListProductInfo struct {
+	ListSourceURL    string                 `json:"listSourceURL"`
+	SoftwareProducts []ChplCertifiedProduct `json:"softwareProducts"`
+}
+
+type ChplCertifiedProduct struct {
+	ChplProductNumber string  `json:"chplProductNumber"`
+	Developer         details `json:"developer"`
+}
+
+type ChplMapResults struct {
+	ChplProductIDs []string
+	ChplDeveloper  string
+}
+
 // MatchEndpointToVendor creates the database association between the endpoint and the vendor,
 // and the endpoint and the healht IT product.
-func MatchEndpointToVendor(ctx context.Context, ep *endpointmanager.FHIREndpointInfo, store *postgresql.Store) error {
+func MatchEndpointToVendor(ctx context.Context, ep *endpointmanager.FHIREndpointInfo, store *postgresql.Store, listSourceMap map[string]ChplMapResults) error {
+
+	fhirEndpointList, err := store.GetFHIREndpointUsingURL(ctx, ep.URL)
+	if err != nil {
+		return errors.Wrap(err, "error getting fhir endpoints from DB")
+	}
+
+	for _, fhirEndpoint := range fhirEndpointList {
+		developerName := listSourceMap[fhirEndpoint.ListSource].ChplDeveloper
+
+		if len(developerName) > 0 {
+			// No errors thrown means a vendor with developer name was found and can be set on ep
+			vendorMatch, err := store.GetVendorUsingName(ctx, developerName)
+			if err != nil {
+				return errors.Wrap(err, "error matching the capability statement to a vendor for endpoint")
+			}
+
+			ep.VendorID = vendorMatch.ID
+			return nil
+		}
+	}
```

Comment:
If the vendor table is empty, this will return an error and then the endpoint information will. not be added to the database (found this out by not running the chpl querier while running make populatedb. Not sure if this should just return nil and not an error if a vendor match is not found?